lib/ruby_lsp/global_state.rb

```diff
@@ -159,6 +159,11 @@ def encoding_name
       end
     end
 
+    sig { returns(T::Boolean) }
```

Comment:
(nit) maybe worth a comment saying this is deprecated, and only for backwards compatibility.

Reply:
Let's get this out first to prevent breaking people's setups.